File: 总题库/121.BestTimeToBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() < 2) return 0;
        int buy = -prices[0], sell = 0;
        for (int i = 1; i < prices.size(); ++i) {
            sell = max(sell, prices[i] - buy)
            buy = max(buy, -prices[i]);
        }
        return sell;
    }
};

File: 总题库/121.BestTimeToBuyandSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2: return 0
        buy, sell = -prices[0], 0
        for i in range(1, len(prices)):
            sell = max(sell, prices[i] + buy)
            buy = max(buy, -prices[i])
        return sell